base/Camera.ts

```diff
@@ -46,7 +46,7 @@ export class Camera {
      * @param duration the duration that the camera will to take to reach target position in ms.
      * @returns returns a promise to the transition end.
      */
-    follow(target?: Camera["target"], duration?: number): Promise<void> {
+    async follow(target?: Camera["target"], duration?: number): Promise<void> {
```

Comment:
Hmm, I'm not sure about the impact of adding this async here... Ok that this function returns a Promise, but perhaps we may face two issues here: adding async will make it return a Promise of Promise? Not sure. Also, not sure whether the synchronized section of this function is expected to finish for some reason in the places where this function is used. Would be nice to check.

Reply:
In my experience, directly `return`ing a `Promise` in an `async` function does not return a `Promise<Promise>`.  But it *would* still be a good idea to double-check and confirm.

Reply:
For our purposes, the above works fine.  https://dmitripavlutin.com/return-await-promise-javascript